aws_lambda_builders/workflows/python_pip/packager.py

```diff
@@ -620,13 +639,79 @@ def _unpack_sdist_into_dir(self, sdist_path, unpack_dir):
         contents = self._osutils.get_directory_contents(unpack_dir)
         return self._osutils.joinpath(unpack_dir, contents[0])
 
-    def get_package_name_and_version(self, sdist_path):
+    def _get_name_version(self, pkg_info_filepath: str) -> Tuple[str, str]:
+        """
+        Extracts the name and the version from the PKG-INFO metadata file
+
+        Parameters
+        ----------
+        pkg_info_filepath: str
+            The path to the PKG-INFO file to get data from
+
+        Returns
+        -------
+        Tuple[str, str]
+            A tuple containing the name and version
+        """
+        metadata = self._parse_pkg_info_file(pkg_info_filepath)
+        return (metadata["Name"], metadata["Version"])
+
+    def _is_default_setuptools_values(self, name: str, version: str) -> bool:
+        """
+        Checks if the name and the version are the default values that are assigned by setuptools
+
+        Parameters
+        ----------
+        name: str
+            The name of the package
+        version: str
+            The version of the package as a string
+
+        Returns
+        -------
+        bool
+            True if either name or version are the default values
+        """
+        # default values logic:
+        # https://github.com/pypa/setuptools/blob/6083e18f4afc40316c0112134c205c336afbcdfd/setuptools/_distutils/dist.py#L1185-L1189
+        return name == "UNKNOWN" or version == "0.0.0"
```

Comment:
Should this be `and`? Doc string indicates `and`.

Reply:
Hmm, thats a good call out. This should be an `or`, so the doc string should be updated. Reason why the version check is also included here is that the assumption is being made that the packages wouldn't have version `0.0.0`. In any case, if it did it would just grab the fall back file (the one `pip download` generates with the build deps) and use what it has.

Reply:
Got it. Probably fine. I did a quick search and seems like most package versions start with `0.1.0` or `0.0.1`. I am sure there are packages with `0.0.0` but we can always handle those in one offs. 

The "safer" thing is probably doing a version check for `0.0.0` only if `UNKNOWN` is given but I think what you have is fine.